notebooks/pixel_classification.ipynb

```diff
@@ -220,67 +228,1850 @@
+      "z: 222/235\n",
+      "z: 223/235\n",
+      "z: 224/235\n",
+      "z: 225/235\n",
+      "z: 226/235\n",
+      "z: 227/235\n",
+      "z: 228/235\n",
+      "z: 229/235\n",
+      "z: 230/235\n",
+      "z: 231/235\n",
+      "z: 232/235\n",
+      "z: 233/235\n",
+      "z: 234/235\n",
+      "z: 235/235\n",
+      "done\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Load the model linked to the dataset\n",
+    "model_file = model_selection.value\n",
+    "\n",
+    "images = conn.getObjects('Image', opts={'dataset': dataset_id})\n",
+    "\n",
+    "# Create a new dataset where to upload the generated images\n",
+    "dataset_obj = omero.model.DatasetI()\n",
+    "v = \"ilastik_probabilities_from_dataset_%s\" % dataset_id\n",
+    "dataset_obj.setName(omero.rtypes.rstring(v))\n",
+    "v = \"ilatisk results probabilities from Dataset:%s\" % dataset_id\n",
+    "dataset_obj.setDescription(omero.rtypes.rstring(v))\n",
+    "dataset_obj = conn.getUpdateService().saveAndReturnObject(dataset_obj)\n",
+    "\n",
+    "# Prepare ilastik\n",
+    "os.environ[\"LAZYFLOW_THREADS\"] = \"2\"\n",
+    "os.environ[\"LAZYFLOW_TOTAL_RAM_MB\"] = \"2000\"\n",
+    "args = app.parse_args([])\n",
+    "args.headless = True\n",
+    "args.project = model_file\n",
+    "shell = app.main(args)\n",
+    "\n",
+    "images = itertools.islice(images, 2)\n",
+    "for image in images:\n",
+    "    filename, file_extension = os.path.splitext(image.getName())\n",
+    "    input_data = load_numpy_array(image)\n",
+    "\n",
+    "    # run ilastik headless\n",
+    "    print('running ilastik using %s and %s' % (model_file, image.getName()))\n",
+    "    role_data_dict = [ {\"Raw Data\": PreloadedArrayDatasetInfo(preloaded_array=input_data, axistags=vigra.defaultAxistags(\"tzyxc\"))}]\n",
+    "\n",
+    "    predictions = shell.workflow.batchProcessingApplet.run_export(role_data_dict, export_to_array=True)\n",
+    "    # Save the probabilities file to the image\n",
+    "    print(\"Saving Probabilities as an Image in OMERO\")\n",
+    "    name = filename + \"_Probabilities\"\n",
+    "    desc = \"ilastik probabilities from Image:%s\" % image.getId()\n",
+    "    for data in predictions:\n",
+    "        # Re-organise array from tzyxc to zctyx order expected by OMERO\n",
+    "        data = data.swapaxes(0, 1).swapaxes(3, 4).swapaxes(2, 3).swapaxes(1, 2)\n",
+    "        conn.createImageFromNumpySeq(plane_gen(), name,\n",
+    "                                     data.shape[0], data.shape[1],\n",
+    "                                     data.shape[2], description=desc,\n",
+    "                                     dataset=dataset_obj)\n",
+    "\n",
+    "print(\"done\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### View the results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "view(data_viewer)"
```

Comment:
``data_viewer`` is not defined in this notebook. I have tried to define it in the previous cell as

```
"    data_viewer = data[:, 0, 0, :, :]\n",
```

inside the ``for`` loop, which stops the error, but still does not draw anything

Reply:
it was an addition from the previous notebook.
I should have rerun the cell
Let me check